File: openapi/code_samples/javascript/api_subscription_creditProfile.js
const express = require('express');
const app = express();
const port = 3000;

app.get('/api/subscription/getActiveSubscriptionForUser', (req, res) => {
    // Your implementation here to retrieve the user's active subscription
    // Simulated data for demonstration purposes
    const userSubscription = {
        userId: "123",
        subscriptionId: "456",
        creditsRemaining: 10
    };

    res.status(200).json(userSubscription);
});

app.listen(port, () => {
    console.log(`Server is listening at http://localhost:${port}`);
});

File: openapi/code_samples/java/api_subscription_creditProfile.java
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/api/subscription")
public class SubscriptionController {

    @GetMapping("/getActiveSubscriptionForUser")
    public ResponseEntity<UserSubscriptionResponse> getActiveSubscriptionForUser() {
        // Your implementation here to retrieve the user's active subscription
        // Simulated data for demonstration purposes
        UserSubscriptionResponse userSubscription = new UserSubscriptionResponse("123", "456", 10);

        return ResponseEntity.ok(userSubscription);
    }
}

@Data
@AllArgsConstructor
class UserSubscriptionResponse {
    private String userId;
    private String subscriptionId;
    private int creditsRemaining;
}
